Task DnneGenerateNativeExports should fail gracefully, and/or provide warnings or errors when no exports are defined

```xml
  <Target
    Name="DnneGenerateNativeExports"
    Condition="('$(DesignTimeBuild)' != 'true' OR '$(BuildingProject)' == 'true') AND '$(DnneBuildExports)' == 'true'"
    Inputs="@(IntermediateAssembly)"
    Outputs="@(DnneGeneratedSourceFile)"
    AfterTargets="CoreCompile">
    <Message Text="Generating source for @(IntermediateAssembly) into @(DnneGeneratedSourceFile)" Importance="$(DnneMSBuildLogging)" />

    <!-- Ensure the output directory exists -->
    <MakeDir Directories="$(DnneGeneratedOutputPath)" />

    <PropertyGroup>
      <DocFlag Condition="Exists($(DocumentationFile))">-d &quot;$(DocumentationFile)&quot;</DocFlag>
    </PropertyGroup>

    <Exec Command="$(DnneGenExe) @(IntermediateAssembly) $(DocFlag) -o @(DnneGeneratedSourceFile)" />
  </Target>
```

The `Exec` task fails and the output stating why it failed requires careful scrutiny of logs.

The build will fail because the output C file (`dnne/*.g.c`) will be absent.
```
c1 : fatal error C1083: Cannot open source file: X:\...\obj\...\dnne\ProjectName.g.c
```

Either it needs to 'allow' the file to be absent, and/or issue warnings or errors instead of simply stating the file is missing.

There are `CustomErrorRegularExpression` and `CustomWarningRegularExpression` which can be applied according to [the Exec task docs](https://learn.microsoft.com/en-us/visualstudio/msbuild/exec-task). Verifying the existence of the output file could also be a good follow up.

--- a/src/msbuild/DNNE.targets
+++ b/src/msbuild/DNNE.targets
@@ -105,12 +105,16 @@ DNNE.targets
 
   <Target
     Name="DnneBuildNativeExports"
-    Condition="'$(DnneBuildExports)' == 'true' AND Exists(@(DnneGeneratedSourceFile))"
+    Condition="'$(DnneBuildExports)' == 'true'"
     Inputs="@(DnneGeneratedSourceFile)"
     Outputs="@(DnneNativeExportsOutputs)"
     AfterTargets="DnneGenerateNativeExports"
     DependsOnTargets="ResolvePackageAssets;ResolveFrameworkReferences">
     <Message Text="Building native exports binary from @(DnneGeneratedSourceFile)" Importance="$(DnneMSBuildLogging)" />
+
+    <Error
+      Condition="!Exists(@(DnneGeneratedSourceFile))"
+      Text="Generated source file(s), @(DnneGeneratedSourceFile), not found. In order to generate source an export must exist. Create an export by adorning a function with UnmanagedCallersOnly." />
 
     <!-- Ensure the output directory exists -->
     <MakeDir Directories="$(DnneGeneratedBinPath)" />

--- a/src/msbuild/DNNE.targets
+++ b/src/msbuild/DNNE.targets
@@ -105,7 +105,7 @@ DNNE.targets
 
   <Target
     Name="DnneBuildNativeExports"
-    Condition="'$(DnneBuildExports)' == 'true'"
+    Condition="'$(DnneBuildExports)' == 'true' AND Exists(@(DnneGeneratedSourceFile))"
     Inputs="@(DnneGeneratedSourceFile)"
     Outputs="@(DnneNativeExportsOutputs)"
     AfterTargets="DnneGenerateNativeExports"

